packages/strapi-design-system/src/Checkbox/Checkbox.tsx

```diff
@@ -1,33 +1,39 @@
-import React from 'react';
+import * as React from 'react';
 
-import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-import { BaseCheckbox } from '../BaseCheckbox';
+import { BaseCheckbox, BaseCheckboxProps } from '../BaseCheckbox';
 import { Box } from '../Box';
 import { Field, FieldHint, FieldError, useField } from '../Field';
 import { Flex } from '../Flex';
 import { useId } from '../hooks/useId';
 import { Typography } from '../Typography';
 
-const CheckboxLabel = styled(Typography)`
+const CheckboxLabel = styled(Typography)<{ disabled: boolean }>`
   display: flex;
   align-items: flex-start;
   * {
     cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
   }
 `;
 
-const CheckboxTick = (props) => {
+const CheckboxTick = (props: BaseCheckboxProps) => {
   const { id } = useField();
 
   return <BaseCheckbox id={id} {...props} />;
 };
 
-export const Checkbox = ({ children, disabled, id, hint, error, ...props }) => {
+interface CheckboxProps extends BaseCheckboxProps {
+  children: React.ReactNode;
+  disabled?: boolean;
+  hint?: string | React.ReactNode | React.ReactNode[];
+  error?: string;
```

Comment:
i think these should just be extended from `FieldProps` imo

Reply:
ok thanks